src/http.cc

```diff
@@ -733,7 +733,7 @@ HttpStateData::processReplyHeader()
     if (!peerSupportsConnectionPinning())
         request->flags.connectionAuthDisabled = true;
 
-    HttpReply *vrep = setVirginReply(newrep);
+    const auto vrep = setVirginReply(newrep.getRaw());
```

Comment:
Please include updating the `setVirginReply()` API and its callers.  All suffer from this same "accidental destruction" flaw in regards to the `HttpReply` object they generate. The fix is the same, they all need to use a local `Pointer` instead of raw-pointer local variable.

Reply:
This would require too much changes, I think...  The 'updating the setVirginReply() API' would require updating Client::theVirginReply to HttpReply::Pointer,  Client::virginReply() getter method and its callers. This, in turn, requires more changes, for otherwise we would need to 'getRaw()' in those contexts...
Note that currently there is no problem with setVirginReply(), because it locks theVirginReply before returning it. Callers may safely lock/unlock the raw pointer it returns.

Reply:
Hmm, the more the better. Though to make some progress I will not insist on this.